python-sdk/src/astro/databases/__init__.py

```diff
@@ -31,11 +32,16 @@ def create_database(conn_id: str, table: BaseTable | None = None) -> BaseDatabas
     :param conn_id: Database connection ID in Airflow
     :param table: (optional) The Table object
     """
+    module = importlib.import_module(_get_conn(conn_id))
+    class_name = get_class_name(module_ref=module, suffix="Database")
+    database: BaseDatabase = getattr(module, class_name)(conn_id, table)
+    return database
+
+
+@cache
```

Comment:
Ohhh this is a fantastic idea. Are you going to do a benchmark of before/after?

Reply:
I wonder what happens if the conn_id doesn't exist anymore.

Reply:
Since this will be called when the task starts execution on the worker -- if the connection exists at that time, we should be fine.

But to answer your question: if `conn_id` existed before but doesn't exist now, it would not throw an error if called again in the same process